File: from-2001-to-2100/2064-minimized-maximum-of-products-distributed-to-any-store/solution.py
class Solution:
    def minimizedMaximum(self, n: int, quantities: List[int]) -> int:
        
        left, right = 1, max(quantities)
        while left < right:
            mid = (left + right) // 2
            if sum(math.ceil(q / mid) for q in quantities) <= n:
                right = mid
            else:
                left = mid + 1
                
        return left
    
    
    def minimizedMaximum_(self, n, Q):
        beg, end = 0, max(Q)

        while beg + 1 < end:
            mid = (beg + end)//2
            if sum(ceil(i/mid) for i in Q) <= n:
                end = mid
            else:
                beg = mid

        return end

File: from-2001-to-2100/2064-minimized-maximum-of-products-distributed-to-any-store/solution.go
import (
    "math"
)

func minimizedMaximum(n int, quantities []int) int {
    low, high := 1, 0
    for _, quantity := range quantities {
        if quantity > high {
            high = quantity
        }
    }
    result := 0

    for low <= high {
        mid := low + (high-low)/2
        storesNeeded := 0
        for _, quantity := range quantities {
            storesNeeded += int(math.Ceil(float64(quantity) / float64(mid)))
        }
        
        if storesNeeded <= n {
            result = mid
            high = mid - 1
        } else {
            low = mid + 1
        }
    }
    
    return result
}